### DISCUSSION TITLE
Can I change the csv to draw other shape (Cesium Entity) ?

### DISCUSSION THREAD

**Discussion by LHKaay** [TerriaJS/terriajs#6796] [Q&A] [answered]
I have a csv file and terria map  draw the point shape only. How to tell the terria map draw other shape (Ex Cesium Entity) and change color base on value from csv file ?

**Comment by na9da** [comment]
Hi @LHKode,

Good question. I'll try to answer them in parts. 

> How to tell the terria map draw other shape (Ex Cesium Entity) 

It is currently not possible to replace a CSV point with custom Cesium entity. 

Although not what you maybe looking for, we can use [CZML](https://github.com/AnalyticalGraphicsInc/czml-writer/wiki/CZML-Guide) for styling GeoJSON/Shape files. [In this example](https://vic.digitaltwin.terria.io/#share=s-n00M0owIgixYDjn3kcGMr9R7Lpl), each tree is a Shapefile point styled as a Cesium billboard image and scaled using its height property. The catalog definition looks something similar to below:

```json
{
  "name": "Trees",
  "type": "geojson",
  "url": "example.org/trees.shp",
  "czmlTemplate": {
    "billboard": {
      "disableDepthTestDistance": 0,
      "height": {
        "reference": "#properties.max_height"
      },
      "heightReference": "CLAMP_TO_GROUND",
      "horizontalOrigin": "CENTER",
      "image": "/images/tree.png",
      "scale": 1,
      "scaleByDistance": {
        "nearFarScalar": [0, 1, 10000, 0.0001]
      },
      "show": true,
      "sizeInMeters": true,
      "verticalOrigin": "BOTTOM",
      "width": {
        "reference": "#properties.diameter"
      }
    }
  }
}
```

Some documentation on the property `czmlTemplate` for geojson is [available here](https://docker-docs--terriajs-docs-v8.netlify.app/guide/connecting-to-data/catalog-type-details/geojson/). I have created a [ticket to investigate](https://github.com/TerriaJS/terriajs/issues/6802) if we can implement the same feature for CSV items.


> change color base on value from csv file 

If you only want to change color or size of the points in a CSV (without using a custom entity). This is currently possible. The exact definition will depend on whether your CSV data is continuous or discrete. [In this example](http://ci.terria.io/main/#share=s-6LoFq7iWWRQI31z8YMyyoWxpM8s), we style individual power stations based on the selected display variable. You can check its [catalog definition here](https://github.com/TerriaJS/nationalmap/blob/9498a138811f9f36e42f2f079369286c3d15c46e/wwwroot/init/simple.json#L253-L279). 

With CSV, it is also possible to use [Maki icons](https://labs.mapbox.com/maki-icons/) to style individual points. Here is an [example](https://nationalmap.gov.au/#share=s-dt0HO6cq8Qugc7MwnBLXjqn8b4y) where I replace each point with a `windmill` icon. Terria has some limited capability to style a CSV (or other vector datasets) from the UI. The following screenshots explain how to [style the points](https://nationalmap.gov.au/#share=s-wKjEMvYkDIMd9P4SO1fCRVCpksz) using an icon through the UI.

1. Open `Style editor` 
   ![image](https://github.com/TerriaJS/terriajs/assets/1430/888c5488-3ea3-4e2e-bc29-b88df8c33172)
2. Choose symbology to edit - `Point/Marker style` 
    ![image](https://github.com/TerriaJS/terriajs/assets/1430/20db6bf4-2ba9-437e-8316-795a3863ab1f)
3. Choose Marker icon 
    ![image](https://github.com/TerriaJS/terriajs/assets/1430/9cec7b72-42e5-4f56-93bc-c0adee39fdbd)
5. Finally you can export the generate catalog definition by clicking the `Show advanced options` menu at the top
    ![image](https://github.com/TerriaJS/terriajs/assets/1430/443e3dbf-625c-401f-a06b-366159f25edd)

    ...and then click again and choose copy-to-clipboard option. This will copy the style definition to clipboard which you can then use in your catalog file.

    ![image](https://github.com/TerriaJS/terriajs/assets/1430/e83bdcef-3683-4eed-a2a1-4aa6298540c6)

Note that the style editor might be a bit hard to understand at first as it was primarily designed as developer tool but it exposes a lot of options.

Happy to help further with any questions. If it is possible, please do share some sample data that you are trying to style.

**Reply by LHKaay** [reply, marked as answer]
Hi @na9da 
Thanks for your useful answer !!! 
It helps me alot. Really look forward to this helpful implementation of this feature.
